### PR TITLE
support setting custom duration on default account, solves #43

### PR DESCRIPTION
**Pull request by arahayrabedian**
So this PR sets a custom duration if defined in the configuration, I'll admit the code is a bit awkward in places (namely passing logger to the static `get_sts_token`), I've not been able to test this with multiple accounts, but it does "workforme" quite thoroughly.

setting too high a duration (in general) and a too high duration for a role gracefully dump an error and exit with `-1`, invalid integer for duration is ignored with a message (and implicitly defaults to 3600)

Happy to re-visit anything that doesn't fit in to the project.

and on that note: thank you for putting `okta-awscli` together, it's so much more efficient/better than the oktadeveloper stuff

**note**: this does *not* optimistically check that a duration is valid for the role

### CODE REVIEW

**Comment by arahayrabedian** [comment]
I'm not sure what the actual terminology for the "okta aws" account would be, but that's what i meant by "default" account, not just "default" configuration

**Comment by arahayrabedian** [comment]
```
# super mega large duration (above AWS limits)
(oktaawscli) ➜  oktaawscli git:(master) ✗ python okta_awscli.py --profile=default -f   
Enter password: 
Enter MFA token: 123456
ERROR - Could not retrieve credentials: 1 validation error detected: Value '22232400' at 'durationSeconds' failed to satisfy constraint: Member must have value less than or equal to 43200

# large duration (above role limits, but within AWS limits)
(oktaawscli) ➜  oktaawscli git:(master) ✗ python okta_awscli.py --profile=default -f
Enter password: 
Enter MFA token: 123456
ERROR - Could not retrieve credentials: The requested DurationSeconds exceeds the MaxSessionDuration set for this role.

# bad integer (letter in the middle)
(oktaawscli) ➜  oktaawscli git:(master) python okta_awscli.py --profile=default -f
Enter password: 
Enter MFA token: 123456
WARNING - Duration could not be converted to a number, ignoring.
```

**Comment by arahayrabedian** [comment]
@jmhale - anything I can do to get this merged in?

**Comment by jmhale** [comment]
@arahayrabedian Sorry for the delay. It's been a crazy week.

Most of it looks good, although I'm seeing some odd behavior where I'm getting back a session expiration time of 1 hour, regardless of what I set the role duration to be.

Example:
I set my test role's max duration to be 2 hours (7200 seconds) and then set the duration option in `~/.okta-aws` to be the same.
I authenticate to the role, but the session expiration time that STS reports back is only 1 hour out from the current time. I added some debug statements to ensure that the specified duration is what is being passed to AWS STS and I'm reasonably sure that it is. I can double-check this by passing a duration that exceeds the max allowed of 7200 and it bombs out, as expected.

The creds that I got back are valid, but I want to sit on them for an hour and see if they are still valid (so, more than one hour, but less than two). I'd like to determine if this is a bug with how AWS is reporting session expiration time or if they're actually invalid after 1 hour.

I would be interested to know if you're seeing the same.

**Comment by jmhale** [comment]
Ok, I think that I figured this out.

You can specify a similar max session duration in the configuration of the Okta app, which gets passed in the SAML assertion. 

If you try to pass a duration to the `sts:AssumeRoleWithSAML` beyond what is is in the assertion, then the actual duration silently falls back to lower value.

I don't think any code change is required here, but we should clearly call out in the README that a admin will need to sufficiently the max session duration on the Okta AWS app itself, as well as in the role config.

@arahayrabedian would you mind updating the README language to call that out? After that, I think this is good to merge.

File: README.md [hunk 1]
```diff
@@ -27,6 +27,7 @@ password = <your_okta_password> # Only save your password if you know what you a
 factor = <your_preferred_mfa_factor> # Current choices are: GOOGLE or OKTA
 role = <your_preferred_okta_role> # AWS role name (match one of the options prompted for by "Please select the AWS role" when this parameter is not specified
 app-link = <app_link_from_okta> # Found in Okta's configuration for your AWS account.
+duration = 3600 # duration in seconds to request a session token for, make sure your account allows for large durations. default: 3600
 ```
```
Comment (jmhale) [inline README.md]: We should have a note here to tell users that the Okta app will need to be configured to allow a larger duration, as well as the AWS role.

**Comment by arahayrabedian** [comment]
@jmhale it's no problem, thanks for getting around to it. I bumped the PR rather optimistically after I started a vacation - made the changes to the readme - let me know if it needs further refinement and I'll do what I can on a phone.

**Comment by jmhale** [comment]
Merging this to develop now. I'm going to sit on cutting the `0.3.0` release for a bit to see if we can get #69 in there too. 

If it's too far out, I'll leave it for another day.

**Comment by arahayrabedian** [comment]
Sweet, thanks @jmhale! - @guerremdq's actually a colleague, I'll give him a ping but no idea if he's also on vacation

**Pull request merged**